tests/test_archiver.py

```diff
@@ -675,3 +676,37 @@ def error(*args, **kwargs):
             self.user,
             [e.message]
         )
+
+class TestArchiverBehavior(OsfTestCase):
+
+    @mock.patch('website.project.model.Node.update_search')
+    def test_archiving_registrations_not_added_to_search(self, mock_update_search):
+        proj = factories.ProjectFactory()
+        reg = factories.RegistrationFactory(project=proj)
+        ArchiveJob(
+            src_node=proj,
+            dst_node=reg,
+            initiator=proj.creator
+        )
+        reg.save()
+        mock_update_search.assert_not_called()
+
+
+    @mock.patch('website.project.model.Node.update_search')
+    @mock.patch('website.archiver.utils.send_archiver_success_mail')
+    def test_archiving_nodes_added_to_search_on_success_if_public(self, mock_send, mock_update_search):
+        proj = factories.ProjectFactory()
+        reg = factories.RegistrationFactory(project=proj)
+        job = ArchiveJob(
+            src_node=proj,
+            dst_node=reg,
+            initiator=proj.creator
+        )
+        reg.save()
+        with nested(
+                mock.patch('website.archiver.model.ArchiveJob.archive_tree_finished', mock.Mock(return_value=True)),
+                mock.patch('website.archiver.model.ArchiveJob.sent', mock.PropertyMock(return_value=False)),
+                mock.patch('website.archiver.model.ArchiveJob.success', mock.PropertyMock(return_value=True))
+        ) as (mock_finished, mock_sent, mock_success):
+            listeners.archive_callback(reg)
+        mock_update_search.assert_called_once_with()
```

Comment:
This isn't actually asserting anything because no expected parameters are being passed. Best to call mock_update_search.assert_called_once() then use `assert_called_once_with` if you want to make assertions about the arguments.